### PR TITLE
remove dfu timeout

### LINKED ISSUE #164: Why boot drive timeout?
With this UF2bootloader (not the Atmel one), there is a timeout after several minutes when the boot drive disappears. Currently this seems to be causing a Finder crash in some circumstances. See https://forums.adafruit.com/viewtopic.php?f=65&t=168987.

@hathach Is the timeout deliberate? What was the motivation? Thanks.

### PR DESCRIPTION
**Pull request by hathach**
fix #164

### CODE REVIEW

**Comment by hathach** [comment]
hmm @dhalbert ci failed to build, complaining the flash is overflow, I am building just fine locally with the very same version of gcc. I just trigger a rebuild of good-old master, and it failed as well. Very weird issue :thinking: :thinking:

File: lib/sdk11/components/libraries/bootloader_dfu/bootloader.c [hunk 1]
```diff
@@ -336,7 +337,9 @@ uint32_t bootloader_dfu_start(bool ota, uint32_t timeout_ms, bool cancel_timeout
       err_code = dfu_transport_ble_update_start();
     }else
     {
-      // timeout_ms > 0 is forced startup DFU
+      // DFU mode with timeout can be
+      // - Forced startup DFU for nRF52832 or
+      // - Makecode single tap reset but no enumerated (battery power)
       if ( timeout_ms )
       {
```
Comment (dhalbert) [inline lib/sdk11/components/libraries/bootloader_dfu/bootloader.c:344]: It looks like you removed the timeout time-keeping code, but why is there still a `timeout_ms` mention here, and mentions of timeout in the comments? I am just a little confused.

**Review by dhalbert — COMMENTED** [review]
I tried this on a CLUE that needed a bootloader update anyway. It works fine. It didn't time out after waiting more than 10 minutes, and it successfully loaded CircuitPython. So my comments are just some questions about the remaining mentions of timeouts in the code.

File: lib/sdk11/components/libraries/bootloader_dfu/bootloader.c (hunk 1 above)
Comment (hathach) [reply lib/sdk11/components/libraries/bootloader_dfu/bootloader.c:344]: This is another timeout, which is different than the overall timeout which is just removed. This one is used by nrf52832, 832 always go into DFU mode for 1 second when power on. This is required for Arduino auto reset is hardware to CP2104 DTR. It is later used by Makecode to exit its singletap reset to DFU when on battery power only (no usb enumeration) after 3 seconds.

Comment (dhalbert) [reply lib/sdk11/components/libraries/bootloader_dfu/bootloader.c:344]: :heavy_check_mark:

**Comment by dhalbert** [comment]
I merged the PR (maybe I should have waited for you), and the build failed: https://github.com/adafruit/Adafruit_nRF52_Bootloader/runs/1075498601

**Comment by hathach** [comment]
> I merged the PR (maybe I should have waited for you), and the build failed: https://github.com/adafruit/Adafruit_nRF52_Bootloader/runs/1075498601

restart the build should fix this, I think it is caused by parallel compiling with `make -j 4`. I experience something similar when compile parallel with `clean all` in local machine. I am not entirely sure though, it was OK until recently. Don't worry I will look at it later on.

**Pull request merged**